packages/vite/src/node/plugins/html.ts

```diff
@@ -850,24 +871,36 @@ export function postImportMapHook(): IndexHtmlTransformHook {
 
 export function resolveHtmlTransforms(
   plugins: readonly Plugin[]
-): [IndexHtmlTransformHook[], IndexHtmlTransformHook[]] {
+): [
+  IndexHtmlTransformHook[],
+  IndexHtmlTransformHook[],
+  IndexHtmlTransformHook[]
+] {
   const preHooks: IndexHtmlTransformHook[] = []
+  const normalHooks: IndexHtmlTransformHook[] = []
   const postHooks: IndexHtmlTransformHook[] = []
 
   for (const plugin of plugins) {
     const hook = plugin.transformIndexHtml
-    if (hook) {
-      if (typeof hook === 'function') {
-        postHooks.push(hook)
-      } else if (hook.enforce === 'pre') {
-        preHooks.push(hook.transform)
+    if (!hook) continue
+
+    if (typeof hook === 'function') {
+      normalHooks.push(hook)
+    } else {
+      const order = hook.order ?? hook.enforce
```

Comment:
@antfu I think this is the only change needed to avoid breaking changed, no? It will still run all `enforce: 'post'` after the vite internal transform, and `order` works as you defined in the PR.
```suggestion
      const order = hook.order ?? (hook.enforce === 'pre' ? 'pre' : undefined)
```

Reply:
Yeah good point! Maybe we could use a comment to explain this?